Assets/LeapMotion/Scripts/VR/LeapVRTemporalWarping.cs

```diff
@@ -182,17 +182,28 @@ protected struct TransformData {
     }
   
     protected void Start() {
-      //Get a callback right as rendering begins for this frame so we can update the history and warping.
-      LeapVRCameraControl.OnValidCameraParams += onValidCameraParams;
-  
-      deviceInfo = provider.GetDeviceInfo();
+
+      Controller controller = provider.GetLeapController();
+      controller.Device += OnDevice;
+    }
+    
+    protected void OnDevice(object sender, DeviceEventArgs args) {
+       deviceInfo = provider.GetDeviceInfo();
       if (deviceInfo.type == LeapDeviceType.Invalid) {
         Debug.LogWarning("Invalid Leap Device -> enabled = false");
         enabled = false;
         return;
       }
+      //Get a callback right as rendering begins for this frame so we can update the history and warping.
+      LeapVRCameraControl.OnValidCameraParams += onValidCameraParams;
```

Comment:
Hmm, your right. It seems the easiest way to avoid double subscription is to unsubscribe before subscribing (since unsubscribing will be ignored if you aren't actually subscribed). Is that too hacky?

Leaving the subscribe in Start() means you would potentially be updating the temporal warping without knowing the device.